### ISSUE TITLE
Classification data is changed when writing copc with non-zero classFlags

### ISSUE THREAD

**Issue by uclaros** [hobuinc/untwine#146]
Untwine seems to modify the _Classification_ attribute for points that have a non-zero ClassFlags.

Given a sample dataset: https://github.com/qgis/QGIS/blob/master/tests/testdata/point_clouds/copc/extrabytes-dataset.copc.laz

`pdal info` shows a Classification attribute range from 2 to 18, and zero for ClassFlags attributes
```
> pdal info extrabytes-dataset.copc.laz
...
      {
        "average": 2.070811158,
        "count": 105972,
        "maximum": 18,
        "minimum": 2,
        "name": "Classification",
        "position": 10,
        "stddev": 0.5567393438,
        "variance": 0.3099586969
      },
...
      {
        "average": 0,
        "count": 105972,
        "maximum": 0,
        "minimum": 0,
        "name": "Synthetic",
        "position": 20,
        "stddev": 0,
        "variance": 0
      },
      {
        "average": 0,
        "count": 105972,
        "maximum": 0,
        "minimum": 0,
        "name": "KeyPoint",
        "position": 21,
        "stddev": 0,
        "variance": 0
      },
      {
        "average": 0,
        "count": 105972,
        "maximum": 0,
        "minimum": 0,
        "name": "Withheld",
        "position": 22,
        "stddev": 0,
        "variance": 0
      },
      {
        "average": 0,
        "count": 105972,
        "maximum": 0,
        "minimum": 0,
        "name": "Overlap",
        "position": 23,
        "stddev": 0,
        "variance": 0
      }
```

If I modify some points to enable eg. the _Synthetic_ flag using a pipeline:
```
[
    "extrabytes-dataset.copc.laz",
    {
        "type": "filters.assign",
        "value": [
            "Synthetic = 1 WHERE GpsTime < 302522581.9720772"
        ]
    },
    "with_class_flags.laz"
]
```
the modification seems successful, _Synthetic_ now has a range of 0 to 1, while everything else is unaffected.
```
> pdal info with_class_flags.laz
...
      {
        "average": 2.070811158,
        "count": 105972,
        "maximum": 18,
        "minimum": 2,
        "name": "Classification",
        "position": 10,
        "stddev": 0.5567393438,
        "variance": 0.3099586969
      },
...
      {
        "average": 0,
        "count": 105972,
        "maximum": 0,
        "minimum": 0,
        "name": "Synthetic",
        "position": 20,
        "stddev": 0,
        "variance": 0
      },
...
```

Passing this file to untwine with `--single_file` creates a copc file that now has altered Classification values:
```
> untwine with_class_flags.copc.laz with_class_flags.laz --single_file
> pdal info with_class_flags.copc.laz
...
{
        "average": 2.080172121,
        "count": 105972,
        "maximum": 34,
        "minimum": 2,
        "name": "Classification",
        "position": 8,
        "stddev": 0.7798070185,
        "variance": 0.6080989861
      },
      {
        "average": 0.0002925301023,
        "count": 105972,
        "maximum": 1,
        "minimum": 0,
        "name": "Synthetic",
        "position": 17,
        "stddev": 0.01710109026,
        "variance": 0.0002924472881
      },
...
```

The Classification values now have a range from 2 to 34, as the modified points got an extra _32_ on their class.

This was tested with the version of untwine bundled in qgis.

**Comment by abellgithub** [comment]
I'm having trouble following the details. If you could simply provide the input, the command you're running and output that you believe is incorrect, that would be helpful.

**Comment by hobu** [comment]
Also which PDAL version? PDAL's flag handling changed in 2.6, and this might have necessitated an update to flag handling in Untwine.

**Comment by uclaros** [comment]
After some thought and re-reading the discussion in https://github.com/PDAL/PDAL/issues/1701 I've changed my mind and am reopening this.

> PDAL isn't just a LAS -> LAS translator. We provide the data to users through an API and users may well be expecting a classification value to be 34, because that's what's in the file.

Untwine _is_ a las -> copc translator, so there is no reason to expect a classification value of 34 when converting from legacy PDRFs.

Minimal example using pdal 2.5, to rule out the changes in ClassFlags handling:
Create a PDRF3 1.2 laz with Synthetic flag, then convert to copc:
```
pdal translate \
https://github.com/qgis/QGIS/raw/master/tests/testdata/point_clouds/copc/extrabytes-dataset.copc.laz \
synthetic_pdrf3.laz \
assign \
--filters.assign.value="ClassFlags=1" 

untwine -i synthetic_pdrf3.laz -o synthetic_pdrf3.copc.laz -s
```

Resulting copc file has Classification values > 31 and ClassFlags properly set.
Expected behavior would be Classification values < 32 and ClassFlags properly set.

**Comment by abellgithub** [comment]
Untwine isn't really a LAS -> copc translator. Untwine accepts any input that PDAL supports. To do what you want, Untwine would need special handling for 1.3 and earlier LAS files. This could be done, but I've also contemplated a dedicated LAS -> COPC program, which would allow for some generally beneficial behavior that can't easily happen in Untwine.

Every time this stuff comes up it's clear that the decision of the LAS committee to save a few bits was not a good one.

**Comment by uclaros** [comment]
Fixed in #148
